magento/magento2
# Improve Bundle Option & Selection Data Extensibility

### Description (*)
Updated the visibility from `private` to `public` for the following functions in `Magento\Bundle\Block\Catalog\Product\View\Type\Bundle` for improved extensibiility:
- `getOptionItemData()`
- `getSelectionItemData()`

### Manual testing scenarios (*)
1. Create a new local module
2. Create a before/after/around plugin for each of the aforementioned functions
3. Configure the plugin in `etc/di.xml`
4. Enable the module
5. Enable and configure Xdebug and add breakpoints in your plugin(s)
6. Load a bundle product page on the frontend
7. Ensure the plugin(s) breakpoint(s) are hit

### Questions or comments
I don't know why these were originally declared `private` functions so if there's a reason this change cannot be made, please let me know.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#34805: Improve Bundle Option & Selection Data Extensibility
